crates/mem-pool/src/sync/mq/gw_kafka.rs

```diff
@@ -0,0 +1,130 @@
+use std::time::Duration;
+
+use anyhow::{anyhow, Result};
+use gw_types::bytes::Bytes;
+use gw_types::packed::{RefreshMemBlockMessage, RefreshMemBlockMessageUnion};
+use gw_types::prelude::{Builder, Entity, Reader};
+use kafka::client::{Compression, FetchOffset, GroupOffsetStorage};
+use kafka::producer::{AsBytes, Record};
+
+use crate::sync::fan_in::FanInMemBlock;
+
+use super::{Consume, Produce};
+
+pub(crate) struct Producer {
+    producer: kafka::producer::Producer,
+    topic: String,
+}
+
+impl Producer {
+    pub(crate) fn new(hosts: Vec<String>, topic: String) -> Result<Self> {
+        let mut client = kafka::client::KafkaClient::new(hosts);
+        client.load_metadata(&[&topic]).map_err(|err| {
+            anyhow!(
+                "topic {:?} not found, load metadata error: {:?}",
+                &topic,
+                err
+            )
+        })?;
+        let producer = kafka::producer::Producer::from_client(client)
+            .with_ack_timeout(Duration::from_secs(1))
+            .with_required_acks(kafka::client::RequiredAcks::One)
+            .with_compression(Compression::SNAPPY)
+            .with_connection_idle_timeout(Duration::from_secs(5))
+            .create()
+            .map_err(|err| anyhow!("Init producer failed: {:?}", err))?;
+
+        Ok(Self { producer, topic })
+    }
+}
+
+impl Produce for Producer {
+    type Msg = RefreshMemBlockMessageUnion;
+
+    fn send(&mut self, message: Self::Msg) -> Result<()> {
+        let msg = RefreshMemBlockMessage::new_builder().set(message).build();
+
+        let bytes = msg.as_bytes();
+        log::trace!("Producer send msg: {:?}", &bytes.to_vec());
+        let message = RefreshMemBlockMessageFacade(bytes);
+        let rec = Record::from_value(&self.topic, message);
+
+        if let Err(err) = self.producer.send(&rec) {
+            log::warn!("Publish message failed, error: {:?}", err);
+        }
+
+        Ok(())
+    }
+}
+
+pub(crate) struct Consumer {
+    consumer: kafka::consumer::Consumer,
+    fan_in: FanInMemBlock,
+}
+
+impl Consumer {
+    pub(crate) fn new(
+        hosts: Vec<String>,
+        topic: String,
+        group: String,
+        fan_in: FanInMemBlock,
+    ) -> Result<Self> {
+        let consumer = kafka::consumer::Consumer::from_hosts(hosts)
```

Comment:
Rename this function to start. The name new implies the function shouldn't has any side-effect